frontend/src/components/pages/acls/UserEditModals.tsx

```diff
@@ -0,0 +1,256 @@
+import {useEffect, useState} from 'react';
+import {generatePassword, StateRoleSelector} from './UserCreate';
+import {useUpdateUserMutation} from '../../../react-query/api/user';
+import {Button, Checkbox, CopyButton, Flex, FormField, IconButton, Modal, ModalBody, ModalContent, ModalFooter, ModalHeader, ModalOverlay, PasswordInput, Tooltip, useToast} from '@redpanda-data/ui';
+import {MdRefresh} from 'react-icons/md';
+import {SingleSelect} from '../../misc/Select';
+import {useListRolesQuery, useUpdateRoleMembershipMutation} from '../../../react-query/api/security';
+import {Features} from '../../../state/supportedFeatures';
+import { create } from '@bufbuild/protobuf';
+import {
+  UpdateRoleMembershipRequestSchema,
+  type UpdateRoleMembershipResponse,
+} from 'protogen/redpanda/api/dataplane/v1/security_pb';
+import { SASLMechanism, UpdateUserRequestSchema } from 'protogen/redpanda/api/dataplane/v1/user_pb';
+import {rolesApi} from '../../../state/backendApi';
+
+type ChangePasswordModalProps = {
+  userName: string;
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+};
+
+export const ChangePasswordModal = ({ userName, isOpen, setIsOpen }: ChangePasswordModalProps) => {
+  const toast = useToast()
+  const [password, setPassword] = useState(generatePassword(30, false));
+  const [mechanism, setMechanism] = useState<SASLMechanism>();
+  const [generateWithSpecialChars, setGenerateWithSpecialChars] = useState(false);
+  const isValidPassword = password && password.length >= 4 && password.length <= 64;
+  const { mutateAsync: updateUser, isPending: isUpdateUserPending } = useUpdateUserMutation();
+
+  const onSavePassword = async () => {
+    const updateRequest = create(UpdateUserRequestSchema, {
+      user: {
+        name: userName,
+        mechanism: mechanism,
+        password: password,
+      },
+    });
+    try {
+      await updateUser(updateRequest);
+      toast({
+        status: 'success',
+        title: `Password for user ${userName} updated`,
+      });
+      setIsOpen(false);
+    } catch (err) {
+      toast({
+        status: 'error',
+        duration: null,
+        isClosable: true,
+        title: `Failed to update user ${userName}`,
+        description: String(err),
+      });
+    }
```

Comment:
This is a connect error, isn't it? Can we use `ConnectError.from(error)` and filter the toast using `formatMessageGrpc` util function?

Reply:
You need to use `formatToastErrorMessageGRPC` to make sure toasts get rendered with consistent text.

Reply:
You can use `showToast(formatToastErrorMessageGRPC(ConnectError.from(error)))` or similar.